pandas/tests/indexing/test_categorical.py

```diff
@@ -807,3 +807,31 @@ def test_loc_with_non_string_categories(self, idx_values, ordered):
         result.loc[sl, "A"] = ["qux", "qux2"]
         expected = DataFrame({"A": ["qux", "qux2", "baz"]}, index=cat_idx)
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "categories",
+        [
+            pytest.param(["a", "b", "c"], id="str"),
+            pytest.param(
+                [pd.Interval(0, 1), pd.Interval(1, 2), pd.Interval(2, 3)],
+                id="pd.Interval",
+            ),
+        ],
+    )
+    def test_categorical_index_reorder_categories(self, categories):
```

Comment:
`test_reorder_index_with_categories`

Reply:
Thanks. Is there a rule (or at least some rough guidance) that would help me choose better test names next time? I couldn't see anything in the style guide, but would be happy to update it with any accepted conventions.

Reply:
No rules but we want it to be as easy as possible to figure out what the test does if/when if fails on a future PR